linker/Mono.Linker.Steps/OutputStep.cs

```diff
@@ -27,15 +27,39 @@
 //
 
 using System;
+using System.Collections.Generic;
 using System.IO;
 
 using Mono.Cecil;
 using Mono.Cecil.Cil;
+using Mono.Cecil.PE;
 
 namespace Mono.Linker.Steps {
 
 	public class OutputStep : BaseStep {
 
+		private Dictionary<UInt16, TargetArchitecture> architectureMap;
+
+		private enum NativeOSOverride {
+			Apple = 0x4644,
+			FreeBSD = 0xadc4,
+			Linux = 0x7b79,
+			NetBSD = 0x1993,
+			Default = 0
+		}
+
+		public OutputStep ()
+		{
+			architectureMap = new Dictionary<UInt16, TargetArchitecture> ();
```

Comment:
Why do you do that when you can compute this on demand in single line?

e.g. `static TargetArchitecture CalculateReadyToRunArchitecture (TargetArchitecture src)`